### PR TITLE
Optimize TCF bundle with just-in-time GVL translations

### PR DESCRIPTION
**Pull request by gilluminate**
Closes [PROD-2281](https://ethyca.atlassian.net/browse/PROD-2281)

### Description Of Changes

In an effort to reduce initial payload size of TCF version of FidesJS, we now ignoring GVL translations in the experience until just as we're rendering the UI.

This is being tightened up by calling the dedicated endpoint for GVL translations.

### Code Changes

* A bit of general code maintenance, including removing a circular dependency in the Location switcher component
* Remove `gvl_translations` from the initial `fides.js` payload.
* When needed, fetch the `gvl_translations` from `/api/v1/plus/gvl/translations` to use just as the UI is rendering.
* Update Cypress and unit tests to reflect the change.

### Steps to Confirm

* Visit demo page with TCF enabled `/fides-js-demo.html?geolocation=fr-id`
* Inspect Network panel of devtools and note that the payload is ~300k smaller than before
* As the page loads, Network panel should show a call to `api/v1/privacy-experience`
* Console debug messages should include `Calling Fides GET GVL translations API...` and `Recieved GVL languages response from Fides API`
* The `Fides has been initialized!` message should appear prior to the `Recieved GVL languages response from Fides API` message in the console. We don't want to delay the overall initialization for this.
* Overlay should display with full expected text (no untranslated strings such as "exp.tcf.stacks.1.name")

image

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

[PROD-2281]: https://ethyca.atlassian.net/browse/PROD-2281?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/fides-js/src/components/tcf/TcfVendors.tsx [hunk 1]
```diff
@@ -281,12 +281,16 @@ const PagedVendorData = ({
...
+      gvlVendors: activeChunk?.filter((v) => v.isGvl),
+      otherVendors: activeChunk?.filter((v) => !v.isGvl),
     }),
     [activeChunk]
   );
 
+  if (!activeChunk) {
+    return null;
+  }
+
```
Comment (gilluminate) [inline clients/fides-js/src/components/tcf/TcfVendors.tsx:293]: handling (unlikely) scenario of having 0 vendors configured for a given experience

File: clients/fides-js/__tests__/__fixtures__/mock_gvl_translations.json [hunk 2]
```diff
@@ -532,7 +532,9 @@
     }
   },
   "es": {
+    "gvlSpecificationVersion": 3,
     "vendorListVersion": 40,
+    "tcfPolicyVersion": 4,
```
Comment (gilluminate) [inline clients/fides-js/__tests__/__fixtures__/mock_gvl_translations.json:537]: resolves a Typescript complaint that these were missing

**Review by eastandwestwind — COMMENTED** [review]
This is on the right track after testing locally. Let's re-test once BE is in place

**Comment by eastandwestwind** [comment]
After https://github.com/ethyca/fidesplus/pull/1510 is merged, lets remove the `gvl_translations: Optional[Dict] = {}` from `src/fides/api/models/privacy_experience.py`?

File: clients/fides-js/src/lib/consent-types.ts [hunk 3]
```diff
@@ -416,6 +416,7 @@ export type PrivacyExperience = {
   gvl?: GVLJson; // NOTE: uses our client-side GVLJson type
   gvl_translations?: GVLTranslations;
```
Comment (eastandwestwind) [inline clients/fides-js/src/lib/consent-types.ts:417]: note to remove this later

File: clients/fides-js/src/services/api.ts [hunk 4]
```diff
@@ -85,12 +87,9 @@ export const fetchExperience = async (
...
     // that have no relevant experiences
-    const experience = (body.items && body.items[0]) ?? {};
-    debugLog(
-      debug,
-      "Got experience response from Fides API, returning: ",
-      experience
-    );
+    const experience: PrivacyExperience = (body.items && body.items[0]) ?? {};
+    debugLog(debug, "Recieved experience response from Fides API", experience);
+    delete experience.gvl_translations;
```
Comment (eastandwestwind) [inline clients/fides-js/src/services/api.ts:92]: note to remove this later

**Review by eastandwestwind — COMMENTED** [review]
Looking good so far, just some notes to remove the `gvl_translations` obj later, including the `gvl_translations: Optional[Dict] = {}` we define on the SQLAlchemy obj

**Review by eastandwestwind — APPROVED** [review]

**Pull request merged**